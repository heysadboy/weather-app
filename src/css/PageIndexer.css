#page-index-container {
    padding: 20px;
}

#left-button {
    float: left;
    color: #424242;
}

#left-button :hover {
    cursor: pointer;
    color: #1b1b1b;
}

#right-button {
    color: #424242;
    float: right;
}

#right-button :hover {
    cursor: pointer;
    color: #1b1b1b;
}