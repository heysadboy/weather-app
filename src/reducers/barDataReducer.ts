import { EActionType } from '../utils/enums';
import { IAction } from '../utils/interfaces';


const barDataReducer = (state = [], action: IAction) => {
    switch (action.type) {
        case EActionType.barchart_data:
            return action.payload;
        default:
            return state;
    }
};

export default barDataReducer;