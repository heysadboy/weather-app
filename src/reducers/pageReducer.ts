import { EActionType } from '../utils/enums';
import { IAction } from '../utils/interfaces';
import { INITIAL_PAGE } from '../utils/data';

const pageReducer = (state = INITIAL_PAGE, action: IAction) => {
    switch (action.type) {
        case EActionType.left_page:
            if (state.start > 0) {
                return { ...state, start: state.start - 1, end: state.end - 1 };
            }
            else {
                return state;
            }

        case EActionType.right_page:
            if (state.end < 5) {
                return { ...state, start: state.start + 1, end: state.end + 1 };
            }
            else {
                return state;
            }
        default:
            return state;
    }
}

export default pageReducer;