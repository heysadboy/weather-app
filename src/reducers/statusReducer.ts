import { INITAIL_STATUS } from '../utils/data';
import { EStatusType } from '../utils/enums';
import { IAction } from '../utils/interfaces';

const statusReducer = (state = {INITAIL_STATUS}, action: IAction) => {
    switch (action.type) {
        case EStatusType.error:
            return action.payload;
        case EStatusType.ok:
            return action.payload;
        case EStatusType.loading:
            return action.payload;
        default:
            return state;
    }
}

export default statusReducer;