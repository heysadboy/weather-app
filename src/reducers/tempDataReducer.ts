import { EActionType } from '../utils/enums';
import { IAction } from '../utils/interfaces';


const tempDataReducer = (state = [], action: IAction) => {
    switch (action.type) {
        case EActionType.temp_data:
            return action.payload;
        default:
            return state;
    }
};

export default tempDataReducer;