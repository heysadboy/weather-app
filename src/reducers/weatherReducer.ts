import { EActionType } from '../utils/enums';
import { IAction } from '../utils/interfaces';


const weatherReducer = (state = {}, action: IAction) => {
    switch (action.type) {
        case EActionType.fetch_weather:
            return action.payload;
        default:
            return state;
    }
};

export default weatherReducer;