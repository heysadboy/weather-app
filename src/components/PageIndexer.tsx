import { FC } from 'react';
import { connect } from 'react-redux';
import { AppState } from '../utils/types'
import { leftPage, rightPage } from '../actions';
import { ThunkDispatch } from 'redux-thunk';
import { bindActionCreators } from 'redux';
import { IAction } from '../utils/interfaces';
import '../css/PageIndexer.css';

interface IPageIndexerProps {
    leftPage: () => void,
    rightPage: () => void
}

const mapDispatchToProps = (
    dispatch: ThunkDispatch<AppState, {}, IAction>
) => ({
    leftPage: bindActionCreators(leftPage, dispatch),
    rightPage: bindActionCreators(rightPage, dispatch)
});

const PageIndexer: FC<IPageIndexerProps> = ({ leftPage, rightPage }) => {

    return (
        <div id="page-index-container">
            <div id="left-button" onClick={() => { leftPage() }}><i className="arrow left big icon" /></div>
            <div id="right-button" onClick={() => { rightPage() }}><i className="arrow right big icon" /></div>
        </div>
    );

}

export default connect(null, mapDispatchToProps)(PageIndexer);