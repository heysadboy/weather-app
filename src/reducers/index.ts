import { combineReducers } from 'redux';
import pageReducer from './pageReducer';
import tempDataReducer from './tempDataReducer';
import weatherReducer from './weatherReducer';

export default combineReducers({
    weather: weatherReducer,
    temparture: tempDataReducer,
    page: pageReducer
});