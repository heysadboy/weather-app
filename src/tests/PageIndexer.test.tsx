import renderer from 'react-test-renderer';
import PageIndexer from '../components/PageIndexer';
import reducers from '../reducers';
import { Provider } from 'react-redux';
import { createStore } from 'redux';


test('Test Page Indexer', () => {
    const store = createStore(reducers);
    const component = renderer.create(
        <Provider store={store}>
            <PageIndexer />
        </Provider>,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});