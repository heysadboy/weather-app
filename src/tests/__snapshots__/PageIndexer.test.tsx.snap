// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Page Indexer 1`] = `
<div
  id="page-index-container"
>
  <div
    id="left-button"
    onClick={[Function]}
  >
    <i
      className="arrow left big icon"
    />
  </div>
  <div
    id="right-button"
    onClick={[Function]}
  >
    <i
      className="arrow right big icon"
    />
  </div>
</div>
`;
